src/UI/Components/Diagram/navigator/navigator.ts

```diff
@@ -0,0 +1,266 @@
+  },
+
+  /**
+   * Sets the tooltip of the element.
+   *
+   * This method is responsible for setting the tooltip of the current element to the provided tooltip.
+   * It also sets the position of the tooltip.
+   *
+   * @param {string} tooltip - The text of the tooltip to set. Defaults to an empty string, which will clear the current tooltip.
+   * @param {string} direction - The position of the tooltip. Defaults to "right".
+   */
+  setTooltip: function (tooltip = "", direction = "right") {
+    this.el.dataset.tooltip = tooltip;
+    this.el.dataset.tooltipPosition = direction;
+  },
+});
+
+/**
+ * NavigatorService class
+ *
+ * This class is responsible for managing the navigation of the canvas.
+ * It provides methods for zoom to fit, zooming in/out in the canvas and fullscreen toggle.
+ *
+ * @class
+ * @property {ui.Toolbar} toolbar - The toolbar object that contains the navigation controls.
+ * @method constructor - Constructs a new instance of the NavigatorService class.
+ * @param {HTMLElement} element - The HTML element that will contain the navigator.
+ * @param {ui.PaperScroller} scroller - The scroller object that allows and zooming in the canvas.
+ */
+export class NavigatorService {
+  toolbar: ui.Toolbar;
+
+  constructor(
+    private element: HTMLElement,
+    private scroller: ui.PaperScroller,
+  ) {
+    this.toolbar = new ui.Toolbar({
+      autoToggle: true,
+      references: {
+        paperScroller: scroller,
+      },
+      tools: [
+        {
+          type: "icon-button",
+          name: "fullscreen",
+        },
+        {
+          type: "icon-button",
+          name: "fit-to-screen",
+          tooltip: "Fit to screen",
+        },
+        {
+          id: "zoomSlider",
+          type: "zoom-slider",
+          min: 0.2 * 100,
+          max: 5 * 100,
+          attrs: {
+            input: {
+              "data-tooltip": "Slide to zoom",
+              "data-tooltip-position": "bottom",
+            },
+          },
+        },
+      ],
+      widgetNamespace: {
+        ...ui.widgets,
+        iconButton: IconButton,
+        // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+      } as any, //ui.widgets aren't aligned with widgetNamespace typing, that's a snippet from the JointJS typeScript demo
+    });
+
+    this.toolbar.render();
+    this.updateFullscreenStyling();
+    this.element.appendChild(this.toolbar.el);
+
+    new ui.Tooltip({
+      rootTarget: ".zoom-handler",
+      target: "[data-tooltip]",
+      padding: 16,
+    });
+
+    const fullscreenButton = this.toolbar.getWidgetByName(
+      "fit-to-screen",
+    ) as IconButton;
+
+    fullscreenButton.setIcon(`${fitToScreen}`);
+
+    this.toolbar.on("fit-to-screen:pointerclick", () => this.fitToScreen());
+    this.toolbar.on("fullscreen:pointerclick", () => this.toggleFullscreen());
+
+    document.addEventListener("fullscreenchange", () =>
+      this.updateFullscreenStyling(),
+    );
+
+    document
```

Comment:
This block here could benefit being split up.
assign the element to a variable and then only add the eventlistener if there's an element.

Also, there's this thing about eventListeners, and anonymous functions, but it might not be applicable in your case.
When you add an anonymous function as eventListener, you can't remove it explicitly, and it will reallocate extra memory on each run because it will redeclare the method. 

Also, you might want to bind your eventHandler method to the context of your class if you want to call it correctly in your addEventListener (It's been a while since I needed that, so I'm not certain)

This is an example on how to bind your functions ;
```ts
class MyClass {
  constructor() {
    this.name = "MyClass Instance";
    this.handleClick = this.handleClick.bind(this); // Bind 'this' to the method
  }

  handleClick(event) {
    console.log(this.name); // 'this' refers to the MyClass instance, not the element
  }

  addEventListenerToElement() {
    const button = document.getElementById('myButton');
    button.addEventListener('click', this.handleClick);
  }
}
```

Reply:
added split and binds, as well as removing events at `remove()`